File: Profile/Pass Task 11.1/coded/Clock/Clock.py
from Counter import Counter

class Clock:
    def __init__(self):
        self.seconds = Counter("Seconds")
        self.minutes = Counter("Minutes")
        self.hours = Counter("Hours")

    def tick(self):
        self.seconds.increment()
        if self.seconds.tick > 59:
            self.minutes.increment()
            self.seconds.reset()  
            if self.minutes.tick > 59:
                self.hours.increment()
                self.minutes.reset()  
                if self.hours.tick > 23:
                    self.reset()

    def reset(self):
        self.seconds.reset()
        self.minutes.reset()
        self.hours.reset()

    def time(self):
        return f"{self.hours.tick:02d}:{self.minutes.tick:02d}:{self.seconds.tick:02d}"

File: Profile/Pass Task 3.3/TestCode/ClockTest/Clock.cs
using System;

namespace ClockTest
{
    public class Clock
    {
        private readonly Counter _seconds = new Counter("seconds");
        private readonly Counter _minutes = new Counter("minutes");
        private readonly Counter _hours = new Counter("hours");

        public void Tick()
        {
            _seconds.Increment();
            if (_seconds.Tick > 59)
            {
                _minutes.Increment();
                _seconds.Reset();
                if (_minutes.Tick > 59)
                {
                    _hours.Increment();
                    _minutes.Reset();
                    if (_hours.Tick > 23)
                    {
                        Reset();
                    }
                }
            }
        }

        public void Reset()
        {
            _seconds.Reset();
            _minutes.Reset();
            _hours.Reset();
        }

        public string Time()
        {
            return $"{_hours.Tick:D2}:{_minutes.Tick:D2}:{_seconds.Tick:D2}";
        }
    }
}
